lib/howl/commands/app_commands.moon

```diff
@@ -302,36 +313,54 @@ command.register
 -----------------------------------------------------------------------
 
 launch_cmd = (working_directory, cmd) ->
-  shell = howl.sys.env.SHELL or '/bin/sh'
-  p = Process {
-    :cmd,
-    :shell,
-    read_stdout: true,
-    read_stderr: true,
-    working_directory: working_directory,
-  }
-
-  buffer = ProcessBuffer p
-  editor = app\add_buffer buffer
-  editor.cursor\eof!
-  buffer\pump!
+  dispatch.launch ->
```

Comment:
Just running each handler in it's own coroutine might be straightforward but I don't think it will solve the issue. The issue stems from having the semantics that the running command 'owns' the command line up until it exits. Only after that are other commands free to take over the command line.

Since the command line state is shared and determined by the stack of running command+interactions, I'm not sure how it would play out if multiple handlers could be active at the same time, each allowed to interact via the command line at will. Consider one command might invoke a yes_or_no question, while another might invoke invoke a read_text in it's own coroutine. Because interactions are on a stack, this would also look weird (':command1_promptcommand2_promptyes_or_no_prompt?read_text_prompt:'). I understand you are not talking about this case, but probably mean that a command that is not going to interact with the command line anymore should not block other commands from running. The issue then is determining that a command is 'not going to interact with the command line anymore'. There is no automatic way to determine this. Yielding is not a good indicator for this because commands may yield while intending to keep ownership of the command line - consider something that reads the file system for a large number of files, or the current replace command that yields to stay responsive while replacing in large buffers.

So the only way I can think of at the moment is for the handler to explicitly declare that it is done with the interactive command line. Other than dispatch.launch (which may ignore errors, as you point out) you can currently use `command_line\run_after_finish` - which could be made to handler errors properly, if it doesn't already. Alternatively, maybe we can have the handler itself return a function to run? E.g.

```moonscript

 handler:
    ...
    return -> launch_cmd working_directory, cmd
```
 
Currently the return value is returned to the caller of `command.run` (but I don't think it's really used anyway) - so we would remove that feature.